test/query_test/query_test.go

```diff
@@ -50,6 +51,81 @@ var _ = Describe("Service Manager Query", func() {
 		}
 	})
 
+	Context("Service instance and last operations query test", func() {
+		var serviceInstance1, serviceInstance2 *types.ServiceInstance
+		BeforeEach(func() {
+			_, serviceInstance1 = common.CreateInstanceInPlatform(ctx, ctx.TestPlatform.ID)
+			_, serviceInstance2 = common.CreateInstanceInPlatform(ctx, ctx.TestPlatform.ID)
+		})
+
+		AfterEach(func() {
+			ctx.CleanupAdditionalResources()
+		})
+
+		It("should return the last operation for a newly created resource", func() {
+			queryParams := map[string]interface{}{
+				"id_list": []string{serviceInstance1.ID},
+			}
+
+			list, err := repository.QueryForList(context.Background(), types.OperationType, storage.QueryForLastOperationsPerResource, queryParams)
+			Expect(err).ShouldNot(HaveOccurred())
+			lastOperation := list.ItemAt(0).(*types.Operation)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(list.Len()).To(BeEquivalentTo(1))
+			Expect(lastOperation.State).To(Equal(types.SUCCEEDED))
+		})
+
+		It("should return the correct last operation after a new one is added to a resource", func() {
+			queryParams := map[string]interface{}{
+				"id_list": []string{serviceInstance1.ID},
+			}
+
+			operation := &types.Operation{
+				Base: types.Base{
+					ID:        "my_test_op_latest",
+					CreatedAt: time.Now(),
+					UpdatedAt: time.Now(),
+					Labels:    make(map[string][]string),
+					Ready:     true,
+				},
+				Description:       "my_test_op_latest",
+				Type:              types.CREATE,
+				State:             types.IN_PROGRESS,
```

Comment:
why IN_PROGRESS? i am worried some maintainer job might try to process it, maybe make it FAILED/SUCCEEDED